apps/design/backend/src/test_decks.ts

```diff
@@ -55,3 +74,194 @@ export function createPrecinctTestDeck({
   });
   return concatenateDocuments(markedBallots);
 }
+
+/**
+ * In order to generate CVRs per sheet, we want to know how contests are
+ * arranged by sheet so we can arrange the votes accordingly.
+ */
+interface BallotContestLayout {
+  precinctId: PrecinctId;
+  ballotStyleId: BallotStyleId;
+  contestIdsBySheet: Array<ContestId[]>;
+}
+
+function getBallotContestLayouts(
+  ballots: BallotLayout[]
+): BallotContestLayout[] {
+  return ballots.map((ballot) => {
+    const { precinctId, gridLayout } = ballot;
+    const { ballotStyleId } = gridLayout;
+    const numSheets = Math.max(
+      ...gridLayout.gridPositions.map((gp) => gp.sheetNumber)
+    );
+    const contestIdsBySheet: BallotContestLayout['contestIdsBySheet'] =
+      Array.from({
+        length: numSheets,
+      }).map(() => []);
+    const oneContestOptionPerContest = uniqueBy(
+      gridLayout.gridPositions,
+      ({ contestId }) => contestId
+    );
+    for (const contestOption of oneContestOptionPerContest) {
+      const { contestId } = contestOption;
+      const { sheetNumber } = contestOption;
+      contestIdsBySheet[sheetNumber - 1].push(contestId);
+    }
```

Comment:
There's probably a nice way to do this using `iter.toMap` to do the grouping, if you're interested in exploring it